ocagent.go

```diff
@@ -370,15 +378,38 @@ func (ae *Exporter) uploadTraces(sdl []*trace.SpanData) {
 		if len(protoSpans) == 0 {
 			return
 		}
+		ae.senderMu.Lock()
```

Comment:
It is odd that we need to use a mutex for a grpc channel send, it should be able to handle concurrent sends and receives with no problems; perhaps is this just for the variable itself that gets reset during reconnection?

Reply:
Afaik gRPC Send isn't concurrency safe (https://groups.google.com/forum/#!topic/grpc-io/aI6L6M4fzQ0), so we need to protect the sends with the mutex.